### DISCUSSION TITLE
Cant load bmp file.

### DISCUSSION THREAD

**Discussion by T0biasCZe** [nothings/stb#1354] [Q&A] [answered]
Hi, i am trying to load 1bit bmp file with the std_image lib, and it returns "process exited with code 0"
When i add the std:cout << stbi_failure_reason, it outputs "bad png sig" (even though its bmp...)
and when i add the #define STBI_ONLY_BMP it outputs "process exited with code -1073741819." (when i remove the failure reason print it returns only "process exited with code 0")
#include <iostream>
//#define STBI_ONLY_BMP
#define STB_IMAGE_IMPLEMENTATION
#include "stb_image.h"

int main(int argc, char* argv[]) {
    int width, height, bpp;
    unsigned char* pixels = stbi_load("C:/Users/user/source/repos/OCBitmapConvert/grayscale.bmp", &width, &height, &bpp, 0);
    //std::cout << stbi_failure_reason();
}

**Comment by T0biasCZe** [comment]
here is the bitmap [grayscale.zip](https://github.com/nothings/stb/files/9228946/grayscale.zip)

**Comment by PathogenDavid** [comment, marked as answer]
You should only call `stbi_failure_reason` when `pixels` is `null`.

(Your image is loading just fine. The error about the PNG signature is a stale message left over from when it checked if the file was a PNG before checking if it was a BMP.)

**Reply by nothings** [reply]
In addition to the above reply, you might also want to learn what a process error code is, because that has nothing to do with what stb_image is doing.
